File: 4045-longest-balanced-subarray-i/longest-balanced-subarray-i.cpp
class Solution {
public:
    int longestBalanced(vector<int>& nums) {
        int n=nums.size();
        int ans=0;
        for(int i=0;i<n;i++){
            set<int>ev,od;
            for(int j=i;j<n;j++){
                if(nums[j]&1) od.insert(nums[j]);
                else ev.insert(nums[j]);
                
                if(ev.size()==od.size()) ans= max(ans,j-i+1);
            }
        }
        return ans;
    }
};

File: 4045-longest-balanced-subarray-i/longest-balanced-subarray-i.java
class Solution {
    public int longestBalanced(int[] nums) {
        int n= nums.length;
        int ans=0;
        for(int i=0;i<n;i++){
        HashSet<Integer>ev=new HashSet<Integer>();
        HashSet<Integer>od=new HashSet<Integer>();
            for(int j=i;j<n;j++){
                if((nums[j]&1)>0) od.add(nums[j]);
                else ev.add(nums[j]);
                if(ev.size()==od.size()) ans=Math.max(ans,j-i+1);
            }
        }
        return ans;
    }
}